test/solidity/Facets/SymbiosisFacet.t.sol

```diff
@@ -0,0 +1,131 @@
+        );
+        symbiosisFacet.setFunctionApprovalBySignature(
+            uniswap.swapExactTokensForETH.selector
+        );
+        symbiosisFacet.setFunctionApprovalBySignature(
+            uniswap.swapETHForExactTokens.selector
+        );
+        symbiosisFacet.setFunctionApprovalBySignature(
+            uniswap.swapTokensForExactETH.selector
+        );
+
+        setFacetAddressInTestBase(address(symbiosisFacet), "SymbiosisFacet");
+
+        bridgeData.bridge = "symbiosis";
+        bridgeData.minAmount = defaultUSDCAmount;
+
+        bytes memory _otherSideCalldata = abi.encodeWithSignature(
+            "synthesize(uint256,address,uint256,address,address,address,address,uint256,bytes32)",
+            1000000, //    bridging fee
+            ADDRESS_USDC, //    token address
+            100000000, //   amount
+            0x0f590DA07186328fCf0Ea79c73bD9b81d3263C2f, //   to,
+            0xb8f275fBf7A959F4BCE59999A2EF122A099e81A8, //    synthesis,
+            0x5523985926Aa12BA58DC5Ad00DDca99678D7227E, //    oppositeBridge,
+            0x0f590DA07186328fCf0Ea79c73bD9b81d3263C2f, //    revertableAddress,
+            56288, //    chainID,
+            "" //    clientID
+        );
+
+        symbiosisData = SymbiosisFacet.SymbiosisData(
+            "", // first swap calldata
+            "", // second swap calldata
+            address(0), //intermediateToken
+            ADDRESS_USDC, // bridgingToken
+            address(0), // firstDexRouter
+            address(0), // secondDexRouter
+            RELAY_RECIPIENT, // bridging entrypoint
+            _otherSideCalldata // core bridging calldata
+        );
+    }
+
+    function initiateBridgeTxWithFacet(bool isNative) internal override {
+        if (isNative) {
+            symbiosisFacet.startBridgeTokensViaSymbiosis{
+                value: bridgeData.minAmount
+            }(bridgeData, symbiosisData);
+        } else {
+            symbiosisFacet.startBridgeTokensViaSymbiosis(
+                bridgeData,
+                symbiosisData
+            );
+        }
+    }
+
+    function initiateSwapAndBridgeTxWithFacet(
+        bool isNative
+    ) internal override {
+        if (isNative) {
+            symbiosisFacet.swapAndStartBridgeTokensViaSymbiosis{
+                value: swapData[0].fromAmount
+            }(bridgeData, swapData, symbiosisData);
+        } else {
+            symbiosisFacet.swapAndStartBridgeTokensViaSymbiosis{
+                value: addToMessageValue
+            }(bridgeData, swapData, symbiosisData);
+        }
+    }
+
+    function testBase_CanBridgeNativeTokens() public override {
+        // native token bridging not supported by core bridge, it's supported by swapping to USDC/WETH on metaRouter
+    }
+
+    function testBase_CanSwapAndBridgeNativeTokens() public override {
+        //native token bridging not supported by core bridge, it's supported by swapping to USDC/WETH on metaRouter
+    }
+}
```

Comment:
Is it possible to add this swapping functionality in the facet?
e.g. If native is received, automatically swap to WETH?